### PR TITLE
DA Verifier service integration tests

### PR DESCRIPTION
**Pull request by bacv**
Verifier service integration tests.

This test mocks a client/zone that encodes data and sends the blobs to multiple nodes to receive attestations.
To run the tests use:
```bash
cargo test -p nomos-da-tests --features libp2p
```

The test that combines verifier, consensus and indexer will be performed when Certificate definition is added to the `kzgrs-backend`.

### CODE REVIEW

**Comment by danielSanchezQ** [comment]
I think the branch is not updated (?) I think I saw a lot of stuff that was already merged.

**Comment by bacv** [comment]
> I think the branch is not updated (?) I think I saw a lot of stuff that was already merged.

🔥 It's fine 🔥

File: nomos-da/kzgrs-backend/src/common/mod.rs [hunk 1]
```diff
@@ -165,6 +171,10 @@ pub fn commitment_to_bytes(commitment: &Commitment) -> Vec<u8> {
     buff.into_inner()
 }
 
-pub struct Attestation {
-    pub signature: Signature,
+fn hash(item: impl AsRef<[u8]>) -> [u8; 32] {
```
Comment (danielSanchezQ) [inline nomos-da/kzgrs-backend/src/common/mod.rs:174]: Lets call it by a proper name `blake2b_hash`.
Also, why using the `Blake2bVar` version for the standard `32bytes` size?

File: nomos-da/kzgrs-backend/src/verifier.rs [hunk 2]
```diff
@@ -131,7 +109,18 @@ impl DaVerifier {
...
+
+        let blob_id = blob.id();
+        let blob_hash: [u8; 32] = blob_id
+            .try_into()
+            .expect("Blob ID must be exactly 32 bytes long");
+
+        Attestation {
+            signature: signature.to_bytes().to_vec(),
+            // TODO: Remove if not required, or implement serialization for 96 bytes pub key.
+            attester: [0u8; 32],
```
Comment (danielSanchezQ) [inline nomos-da/kzgrs-backend/src/verifier.rs:121]: This is probably not needed? The one receiving attestation should know how is answering. Usually you do not want to check a signature over a pk that comes in the same message.

File: nomos-services/data-availability/verifier/src/lib.rs [hunk 3]
```diff
@@ -98,9 +100,8 @@ where
 
 impl<Backend, N, S> ServiceData for DaVerifierService<Backend, N, S>
 where
-    Backend: VerifierBackend,
+    Backend: VerifierBackend<Error = DynError>,
     Backend::Settings: Clone,
-    Backend::Error: Error,
```
Comment (danielSanchezQ) [inline nomos-services/data-availability/verifier/src/lib.rs:103]: Why this error changes? We are going to a more specific type. Is the generic not enough? I think expanding the bound of the generic to what's needed would have been better.

File: nomos-services/data-availability/tests/src/verifier_integration.rs [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+        // Create cert
+        let a1 = node1_reply_rx.await.unwrap();
+        let a2 = node2_reply_rx.await.unwrap();
+
+        if a1.is_some() && a2.is_some() {
+            is_success_tx.store(true, SeqCst);
+        }
+
+        // TODO: Create cert and check indexer integration.
+        performed_tx.store(true, SeqCst);
```
Comment (danielSanchezQ) [inline nomos-services/data-availability/tests/src/verifier_integration.rs:294]: We will check with the indexer to see if the blob is available right? Gonna come in a different PR? (If so lets open an issue I think).

File: nomos-services/data-availability/verifier/src/lib.rs (hunk 3 above)
Comment (bacv) [reply nomos-services/data-availability/verifier/src/lib.rs]: To keep the error generic I've created a new verifier backend error. My idea was that DynError is generic enough for the service.

File: nomos-da/kzgrs-backend/src/common/mod.rs (hunk 1 above)
Comment (bacv) [reply nomos-da/kzgrs-backend/src/common/mod.rs]: Removed this method and used Blake2b (default 32) in the attestation directly.

File: nomos-da/kzgrs-backend/src/common/blob.rs [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+use crate::common::Column;
+use crate::common::Commitment;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DaBlob {
+    pub column: Column,
+    #[serde(
+        serialize_with = "serialize_commitment",
+        deserialize_with = "deserialize_commitment"
+    )]
```
[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply nomos-da/kzgrs-backend/src/common/blob.rs]: That is much better. Thanks!

**Review by danielSanchezQ — APPROVED** [review]
Nice stuff! Thanks!

**Pull request merged**